entities/sync/sync_test.go

```diff
@@ -70,6 +81,185 @@ func TestKeyLockerLockUnlock(t *testing.T) {
 	r.False(mutexLocked(lock.(*sync.Mutex)))
 }
 
+func TestKeyLockerContextMutexLockUnlock(t *testing.T) {
+	r := require.New(t)
+	s := NewKeyLockerContext()
+
+	s.Lock("t1")
+	lock, _ := s.m.Load("t1")
+	r.True(contextMutexLocked(lock.(*contextMutex)))
+
+	s.Unlock("t1")
+	lock, _ = s.m.Load("t1")
+	r.False(contextMutexLocked(lock.(*contextMutex)))
+
+	s.TryLockWithContext("t2", context.Background())
+	lock, _ = s.m.Load("t2")
+	r.True(contextMutexLocked(lock.(*contextMutex)))
+
+	s.Unlock("t2")
+	lock, _ = s.m.Load("t2")
+	r.False(contextMutexLocked(lock.(*contextMutex)))
+}
+
+// Lock a key, then try to lock it concurrently from multiple goroutines with a context that gets cancelled
+// all should give up after cancellation and return false from TryLockWithContext
+func TestKeyLockerContextMutexLockConcurrentCancel(t *testing.T) {
+	s := NewKeyLockerContext()
+	s.Lock("t1")
+	defer s.Unlock("t1")
+
+	numGoroutines := 10
+	ctx, cancel := context.WithCancel(context.Background())
+	wgOuter := sync.WaitGroup{}
```

Comment:
`wgOuter` does not seem needed. goroutines are started synchronously, there is nothing to wait for. (also applies to other tests)

Reply:
still there ;)